tree/treeplayer/src/TTreeFormula.cxx

```diff
@@ -2961,8 +2961,10 @@ Int_t TTreeFormula::DefinedVariable(TString &name, Int_t &action)
             if (current[0] == ']') {
                fIndexes[code][dim] = -1; // Loop over all elements;
             } else {
+               TString tempIndex(current);
+               tempIndex.Remove(TString::kTrailing, ' ').Remove(TString::kTrailing, ']');
```

Comment:
The 2nd remove is suspicious ...

Reply:
Why? current is sth like "2-1]"

Reply:
I had missing the part where this is being done on a copy.  Fair enough as is, I guess.